src/generic/trilinos_eigen_solver.h

```diff
@@ -765,6 +854,116 @@ namespace oomph
+                                    Vector<DoubleVector>& eigenvector)
+    {
+      // No access to sigma, so set from sigma real
+      Sigma = Sigma_real;
+
+      // Initially be dumb here
+      Linear_solver_pt = problem_pt->linear_solver_pt();
+
+      // Let's make the initial vector
+      Teuchos::RCP<DoubleMultiVector> initial = Teuchos::rcp(
+        new DoubleMultiVector(1, problem_pt->dof_distribution_pt()));
+      Anasazi::MultiVecTraits<double, DoubleMultiVector>::MvRandom(*initial);
+
+      // Make the operator
+      // NB Using AdjointProblemBasedShiftInvertOperator, the only difference
+      // to solve_eigenproblem
+      Teuchos::RCP<DoubleMultiVectorOperator> Op_pt =
+        Teuchos::rcp(new AdjointProblemBasedShiftInvertOperator(
+          problem_pt, this->linear_solver_pt(), Sigma));
+
+      // Create the basic problem
+      Teuchos::RCP<Anasazi::BasicEigenproblem<double,
+                                              DoubleMultiVector,
+                                              DoubleMultiVectorOperator>>
+        anasazi_pt = Teuchos::rcp(
+          new Anasazi::BasicEigenproblem<double,
+                                         DoubleMultiVector,
+                                         DoubleMultiVectorOperator>(Op_pt,
+                                                                    initial));
+
+      // Think I have it?
+      anasazi_pt->setHermitian(false);
+
+      // set the number of eigenvalues requested
+      anasazi_pt->setNEV(n_eval);
+
+      // Inform the eigenproblem that you are done passing it information
+      if (anasazi_pt->setProblem() != true)
+      {
+        oomph_info << "Anasazi::BasicEigenproblem::setProblem() had an error."
+                   << std::endl
+                   << "End Result: TEST FAILED" << std::endl;
+      }
+
+      // Create the solver manager
+      // No need to have ncv specificed, Triliinos has a sensible default
+      //  int ncv = 10;
+      MT tol = 1.0e-10;
+      int verbosity =
+        Anasazi::Warnings + Anasazi::FinalSummary + Anasazi::TimingDetails;
+
+      Teuchos::ParameterList MyPL;
+      MyPL.set("Which", "LM");
+      MyPL.set("Block Size", 1);
+      //  MyPL.set( "Num Blocks", ncv );
+      MyPL.set("Maximum Restarts", 500);
+      MyPL.set("Orthogonalization", "DGKS");
+      MyPL.set("Verbosity", verbosity);
+      MyPL.set("Convergence Tolerance", tol);
+      Anasazi::BlockKrylovSchurSolMgr<double,
+                                      DoubleMultiVector,
+                                      DoubleMultiVectorOperator>
+        BKS(anasazi_pt, MyPL);
+
+      // Solve the problem to the specified tolerances or length
+      Anasazi::ReturnType ret = BKS.solve();
+      bool testFailed = false;
+      if (ret != Anasazi::Converged)
+      {
+        testFailed = true;
+      }
+
+      if (testFailed)
+      {
+        oomph_info << "Eigensolver not converged\n";
+      }
```

Comment:
Why not combine this? I.e.
```cpp
if (ret != Anasazi::Converged)
{
   oomph_info << "Eigensolver not converged\n";
}
```
as `testFailed` is a local variable and doesn't appear to be used for the rest of the function.

Reply:
Agreed!

Reply:
Also agree, may be left over from a different usage case where it was reused.

Reply:
Done